conans/test/integration/workspace_test.py

```diff
@@ -80,166 +64,255 @@ def package_info(self):
+        self.assertIn("Hello World C Release!", client.out)
+        self.assertIn("Hello World B Release!", client.out)
+        self.assertIn("Hello World A Release!", client.out)
+
+        build_type = "Debug"
+        client.run("build C -bf=C/build/%s" % build_type)
+        client.run("build B -bf=B/build/%s" % build_type)
+        client.run("build A -bf=A/build/%s" % build_type)
+
+        client.runner(cmd_debug, cwd=client.current_folder)
+        self.assertIn("Hello World C Debug!", client.out)
+        self.assertIn("Hello World B Debug!", client.out)
+        self.assertIn("Hello World A Debug!", client.out)
+
+    def complete_single_conf_build_test(self):
+        client = TestClient()
+
+        def files(name, depend=None):
+            includes = ('#include "hello%s.h"' % depend) if depend else ""
+            calls = ('hello%s();' % depend) if depend else ""
+            deps = ('"Hello%s/0.1@lasote/stable"' % depend) if depend else "None"
+            return {"conanfile.py": conanfile_build.format(deps=deps, name=name),
+                    "src/hello%s.h" % name: hello_h.format(name=name),
+                    "src/hello.cpp": hello_cpp.format(name=name, includes=includes, calls=calls),
+                    "src/CMakeLists.txt": cmake.format(name=name)}
+
+        client.save(files("C"), path=os.path.join(client.current_folder, "C"))
+        client.save(files("B", "C"), path=os.path.join(client.current_folder, "B"))
+        A = os.path.join(client.current_folder, "A")
+        a = files("A", "B")
+        a["src/CMakeLists.txt"] += ("add_executable(app main.cpp)\n"
+                                    "target_link_libraries(app helloA)\n")
+        a["src/main.cpp"] = main_cpp
+        client.save(a, path=A)
+
+        project = dedent("""
+            HelloB/0.1@lasote/stable:
+                folder: B
+            HelloC/0.1@lasote/stable:
+                folder: C
+            HelloA/0.1@lasote/stable:
+                folder: A
+            layout: layout
+            generator: cmake
+            root: HelloA/0.1@lasote/stable
+            """)
+        layout = dedent("""
+            [folders]
+            build: build/{settings.build_type}
+            src: src
+
+            [includedirs]
+            src
+
+            [libdirs]
+            build/{settings.build_type}/lib
+            """)
+        metacmake = dedent("""
+            cmake_minimum_required(VERSION 3.3)
+            project(MyProject CXX)
+            include(${CMAKE_BINARY_DIR}/conanworkspace.cmake)
+            """)
+        client.save({"conanws.yml": project,
+                     "layout": layout,
+                     "CMakeLists.txt": metacmake})
+
+        base_release = os.path.join(client.current_folder, "build_release")
+        base_debug = os.path.join(client.current_folder, "build_debug")
+
+        with client.chdir("build_release"):
+            client.run("workspace install ../conanws.yml")
```

Comment:
Can we run the first part of the test inside this context and the second one inside the `build_debug` one? Also, why not parameterize?

Reply:
I have tried, but there are some lines in between that doesn't run in that folder, and makes it a bit cluttered with a lot of ``with client.chdir``. So leaving this as is.

Not parameterize, because the important thing here is that they run in the same execution, both debug and release might interfere with each other. They did, I solved a lot of bugs because the test runs both in the same execution.